eng/PackageDownloadAndReference.targets

```diff
@@ -0,0 +1,18 @@
+<Project>
+
+  <ItemDefinitionGroup>
+    <PackageDownloadAndReference>
+      <Folder>lib/$(TargetFramework)</Folder>
+      <AssemblyName>%(Identity)</AssemblyName>
+      <Private>false</Private>
+    </PackageDownloadAndReference>
+  </ItemDefinitionGroup>
+
+  <ItemGroup>
+    <PackageDownload Include="@(PackageDownloadAndReference)" />
+    <PackageDownload Update="@(PackageDownloadAndReference)" Version="[%(Version)]"/>
+    <PackageDownloadAndReference Update="@(PackageDownloadAndReference)" PackageFolder="$([System.String]::new(%(Identity)).ToLowerInvariant())" />
+    <Reference Include="@(PackageDownloadAndReference->'$(NuGetPackageRoot)%(PackageFolder)/%(Version)/%(Folder)/%(AssemblyName).dll')" />
```

Comment:
Won't this approach break the ability to use CPM and transitive pinning?

Reply:
This doesn't pull any transitive dependencies at all.  That's the point.  It's a stand-in for a targeting pack.  We pull down just single package and use it only for reference.  It's very manual, but it avoids accidentally lifting dependencies that aren't in the control of the plugin (or pulling down dependencies just to drop them in favor of a framework library).  It should be used sparingly as it is bypassing nuget resolution.

Reply:
Sure but this means it's a different kind of package reference that you can't manage with CPM correct?

Reply:
We'd just use the same properties when setting these that we'd feed into CPM.  It's a temporary solution.  We need an official package or SDK from MSBuild (and roslyn) for targeting hosts.